app/views/questionnaire.py

```diff
@@ -378,6 +379,15 @@ def update_questionnaire_store_with_answer_data(questionnaire_store, location, a
         questionnaire_store.completed_blocks.append(location)
 
 
+def _is_invalid_form(form):
+    is_valid = form.validate()
+    if form.csrf_token.errors:
+        logger.warn('Invalid csrf token')
+        raise Forbidden
```

Comment:
This exception doesn't appear to be caught by our errors view to render our happy error pages.